src/test/ml-data/feature-services/test/DataSourceArray.json

```diff
@@ -469,6 +469,246 @@
         }
       },
       "readOnly": true
+    },
+    {
+      "id": 12,
+      "name": "GKG Left Outer Join",
+      "description": "GDelt article data with Left Outer Join",
+      "geometryType": "Point",
+      "idField": "OBJECTID",
+      "displayField": "name",
+      "extent": {
+        "xmin": -180,
+        "ymin": -90,
+        "xmax": 180,
+        "ymax": 90,
+        "spatialReference": {
+          "wkid": 4326,
+          "latestWkid": 4326
+        }
+      },
+      "dataSources": [
+        {
+          "source": "view",
+          "schema": "GDeltGKG",
+          "view": "Article",
+          "fragmentIdColumn": "FRAGMENT_ID_Article",
+          "joinFunction": "joinLeftOuter",
+          "fields": {
+            "OBJECTID": {
+            "scalarType": "long"
+            }
+          },
+          "includeFields": [
+            "OBJECTID"
+          ]
+        },
+        {
+          "source": "view",
+          "schema": "GdeltGKG",
+          "view": "Article_Join",
+          "fragmentIdColumn": "FRAGMENT_ID_Article_Join",
+          "joinFunction": "joinLeftOuter",
```

Comment:
Can this be put inside the `joinOn` object? Maybe 
```
joinOn {
  left
  right
  joinFunction
}
```
Maybe `joinFunction` could be just `function` if inside the `joinOn` object?